homeassistant/components/sanix/__init__.py

```diff
@@ -0,0 +1,40 @@
+"""The Sanix integration."""
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONF_SERIAL_NO, CONF_TOKEN, DOMAIN
+from .coordinator import SanixCoordinator
+from .sanix import Sanix
+
+PLATFORMS: list[Platform] = [Platform.SENSOR, Platform.BINARY_SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Sanix from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    _serial_no = entry.data[CONF_SERIAL_NO]
+    _token = entry.data[CONF_TOKEN]
+
+    sanix_api = Sanix(_serial_no, _token, async_get_clientsession(hass))
+    coordinator = SanixCoordinator(hass, sanix_api)
+    hass.data[DOMAIN][entry.entry_id] = coordinator
+
+    hass.config_entries.async_update_entry(entry, unique_id=_serial_no)
+
+    await coordinator.async_config_entry_first_refresh()
```

Comment:
Please only assign the coordinator to hass.data after this line

Reply:
Should this `await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)` be before the `async_config_entry_first_refresh`?

Reply:
Nope.
Create coordinator -> first_refresh -> hass.data[...] = coordinator -> forward_entry_setups

Reply:
Okay, so now it looks good.